supybot_fedora/plugin.py

```diff
@@ -615,84 +683,117 @@ def fasinfo(self, irc, msg, args, name):
         """<username>
 
         Return information on a Fedora Account System username."""
-        try:
-            person = self.fasclient.person_by_username(name)
-        except Exception:
-            irc.reply('Error getting info for user: "%s"' % name)
-            return
+
+        person = self._get_person_by_username(irc, name)
         if not person:
-            irc.reply('User "%s" doesn\'t exist' % name)
             return
-        person["creation"] = person["creation"].split(" ")[0]
-        string = (
-            "User: %(username)s, Name: %(human_name)s"
-            ", email: %(email)s, Creation: %(creation)s"
-            ", IRC Nick: %(ircnick)s, Timezone: %(timezone)s"
-            ", Locale: %(locale)s"
-            ", GPG key ID: %(gpg_keyid)s, Status: %(status)s"
-        ) % person
-        irc.reply(string.encode("utf-8"))
 
-        # List of unapproved groups is easy
-        unapproved = ""
-        for group in person["unapproved_memberships"]:
-            unapproved = unapproved + "%s " % group["name"]
-        if unapproved != "":
-            irc.reply("Unapproved Groups: %s" % unapproved)
-
-        # List of approved groups requires a separate query to extract roles
-        constraints = {"username": name, "group": "%", "role_status": "approved"}
-        columns = ["username", "group", "role_type"]
-        roles = []
-        try:
-            roles = self.fasclient.people_query(
-                constraints=constraints, columns=columns
+        irc.reply(str(self.fasjsonclient.operations))
+        if self.registryValue("use_fasjson"):
+            irc.reply(
+                f"User: {person.get('username')}, "
+                f"Name: {person.get('human_name')}, "
+                f"Email: {' and '.join(e for e in person['emails'] or ['None'])}, "
+                f"Creation: {person.get('creation')}, "
+                f"IRC Nicks: {' and '.join(n for n in person['ircnicks'] or ['None'])}, "
+                f"Timezone: {person.get('timezone')}, "
+                f"Locale: {person.get('locale')}, "
+                f"GPG Key IDs: {' and '.join(k for k in person['gpgkeyids'] or ['None'])}, "
+                f"Status: {person.get('status')}"
             )
-        except Exception:
-            irc.reply("Error getting group memberships.")
-            return
 
-        approved = ""
-        for role in roles:
-            if role["role_type"] == "sponsor":
-                approved += "+" + role["group"] + " "
-            elif role["role_type"] == "administrator":
-                approved += "@" + role["group"] + " "
-            else:
-                approved += role["group"] + " "
-        if approved == "":
-            approved = "None"
+            groups = self.fasjsonclient.list_user_groups(username=name).result
+            irc.reply(f"Groups: {', '.join(g['groupname'] for g in groups)}")
```

Comment:
We don't return the role type with the user groups, so we can't add the `+` or `@` prefixes, do you think we should add it to FASJSON?

Reply:
Okies, i will file issues here and in FASJSON to track this, and add at a future date.